### ISSUE TITLE
LuLu causing protocol errors in MacOS Sequoia? 

### ISSUE THREAD

**Issue by jguerin** [objective-see/LuLu#630]
I'm on MacOS 15.0 Beta (24A5320a) with LuLu 2.6.3 and I keep finding that I'm getting SSL protocol errors if LuLu is running, but none if I disable. Is this a known issue?

**Comment by xiaoxchou** [comment]
The same issue appeared on 15.0 Beta(24A5327a)

**Comment by objective-see** [comment]
It's common that 3rd-party software has issues in the new OS betas, as  generally the OS betas themselves have myriad of issues. As such I usually hold off testing until the betas reach the RC stage (especially as LuLu is running fine on the previous version of macOS).

**Comment by jguerin** [comment]
@objective-see There's an understanding that this build is the final before RC gets released, so I wouldn't expect too many more changes?

**Comment by alcortazzo** [comment]
I'm also on RC 15.0 (24A335) right now, and the only networking issues I've found so far are with Slack huddles stability.
Disabling lulu solves the issues.

**Comment by wastez** [comment]
I can confirm the same problem, lost connection while using ssh or using rdp.

With ssh i get the following error since macOS 15:
<img width="593" alt="Bildschirmfoto 2024-09-10 um 19 07 03" src="https://github.com/user-attachments/assets/62178dc7-108b-4b85-8c93-ad56a2a98640">

**Comment by dcb-l** [comment]
Also experiencing issues across most applications with LuLu running on 15RC. The only thing that works reliably is Microsoft Remote Desktop accessing a local network host. Otherwise there are issues with all browsers, VMware Horizon Client, Samba, SSH, Teams, etc with LuLu running.

**Comment by nimishck** [comment]
I started facing this issue as soon as I upgraded to the public release of MacOS Sequoia.

**Comment by objective-see** [comment]
Thanks for feedback and info on this 🙏🏽

If any of you have any relevant log message (either from the system log, or from lulu's log (which can be streamed via:  `log stream --level debug --predicate="subsystem='com.objective-see.lulu'"`) and could share those here, that would be super helpful. 

Also, are there any other system extensions installed? 
(View installed extensions via: `systemextensionsctl list`)

Digging into this more, but as everything seemed to be working fine on macOS 14.*, a bit tough to know where to dig! 😟

**Comment by nimishck** [comment]
I did not see anything in the log at the time the SSH connection was dropped.

Here's the error I get on the SSH side in iTerm:
```
Bad packet length 2860171220.
ssh_dispatch_run_fatal: Connection to <IP-Address> port 22: Connection corrupted
```
It seems like LuLu is interfering with the encrypted packets in some way. When LuLu is disabled, this error does not occur. With LuLu running, the SSH connection does not last for more than 30 seconds or so. The issue seems to be worse with SSH proxy connections using the `ssh -J` option.

Other than LuLu, I have the iTop VPN extension. But that is disabled.

**Comment by wastez** [comment]
The problem seems to be related to the internal firewall of sequoia.

Just disable the internal firewall until apple will fix that.

**Comment by jguerin** [comment]
Yeah, the issue happens with Defender's network filter too. I'm updating to MacOS 15.1 Beta, so will report back if it still happens, then try disabling the firwall.

**Comment by jguerin** [comment]
MacOS 15.1 Beta 4. Firewall, LuLu and Defender network filters enabled, have yet to see a network glitch. 
<img width="498" alt="image" src="https://github.com/user-attachments/assets/13719885-0cc2-4bbe-9ec8-f46f8ed0379e">
<img width="491" alt="image" src="https://github.com/user-attachments/assets/77893c6c-e608-47db-8315-e2529c730a73">

Must've been an OS bug. Will leave this open for others to confirm.

**Comment by alcortazzo** [comment]
> The problem seems to be related to the internal firewall of sequoia.
> 
> Just disable the internal firewall until apple will fix that.
> 
> The problem happens too without lulu. (But it is a lot rarer) A normal user would not get it but related to my work i'm connected to ssh the hole day.

I can confirm that after enabling Lulu and disabling the internal firewall, everything seems to be working fine.

**Comment by peterneutron** [comment]
I can also confirm that the issues are gone after deactivating the internal firewall.

**Comment by objective-see** [comment]
This has now been confirmed, to yes, be due to an Apple bug in macOS 15, that is widely impacting many 3rd-party security tools, that then in turn causes macOS networking to break.

This has been reported to Apple (and was so **before** macOS 15 was released), who have now confirmed the issue and are hopefully working on a fix.

More info: "[Apple’s new macOS Sequoia update is breaking some cybersecurity tools](https://techcrunch.com/2024/09/19/apples-new-macos-sequoia-update-is-breaking-some-cybersecurity-tools/)"

**Comment by jguerin** [comment]
Yeah, I think that someone from my company tried to report it, but as you highlighted in your Twitter post, Apple isn't super responsive to these kinds of feedback, even when coming from large IT companies.

**Comment by jguerin** [comment]
I'm on the latest 15.1 Beta with firewall and all network filters enabled and am no longer experiencing network issues.

**Comment by yakirlog** [comment]
I'm having issues too.. Mainly with connection & transferring file to my NAS..
So what to do in the meanwhile? should I disable LuLu only? macOS firewall only? both?

**Comment by objective-see** [comment]
@yakirlog you can either disable the macOS firewall (which blocks incoming connections), **or** LuLu (which blocks unauthorized outgoing connections).

**Comment by wastez** [comment]
@objective-see 
Also if lulu is disabled it can also happen. (much less often but still happening)

**Comment by objective-see** [comment]
@yakirlog ah good to know, and makes sense as its yes, a macOS bug. In that case, seems like disabling the internal firewall is the way to go. Good news is, macOS 15.1 should be out shortly with fixes from Apple 🤞🏽

**Comment by myndzi** [comment]
I have been encountering the same problem since I updated. I tried the beta, but contrary to other posts, it does _not_ seem resolved on the beta, though it does seem potentially less frequent.

I watch a lot of Twitch, and it shows up as streaming video interruptions and chat disconnections. Disabling Lulu does seem to clear it up.

**Comment by reedloden** [comment]
macOS 15.0.1 [out today](https://www.macrumors.com/2024/10/03/apple-releases-macos-sequoia-15-0-1-with-bug-fixes/) with a fix that "Improves compatibility with third-party security software".

**Comment by jpap** [comment]
> macOS 15.0.1 [out today](https://www.macrumors.com/2024/10/03/apple-releases-macos-sequoia-15-0-1-with-bug-fixes/) with a fix that "Improves compatibility with third-party security software".

I'm still seeing this on 15.0.1 (arm64) with LuLu 2.6.3. :(

It's easy to reproduce: just SSH into a Linux box and run `top` until the connection gets corrupted:

```
Bad packet length 2202000445.
ssh_dispatch_run_fatal: Connection to XYZ port 22: Connection corrupted
```

**Comment by peterneutron** [comment]
Seems like Apple is on a roll recently, between this and all the updates they pulled that ended up bricking Watches and M4 iPads.

**Comment by myndzi** [comment]
It seemed possibly to be worse when my computer was under load, so I wonder if it's something similar to like, dropping frames in a game when you can't do the work in time. Except the work here is ... ???

Since it seems to affect network filters, I could potentially see it like a timeout waiting for the filter to complete its work. But if it's actually _corrupting_ the network data stream, then I'd be more concerned it's actually a potential security problem, a buffer getting overwritten/mangled or something like that

**Comment by peterneutron** [comment]
> > macOS 15.0.1 [out today](https://www.macrumors.com/2024/10/03/apple-releases-macos-sequoia-15-0-1-with-bug-fixes/) with a fix that "Improves compatibility with third-party security software".
> 
> I'm still seeing this on macOS 15.0.1 (arm64) with LuLu 2.6.3. :(
> 
> It's easy to reproduce: just SSH into a remote machine and run `top` until the connection gets corrupted:
> 
> ```
> Bad packet length 2202000445.
> ssh_dispatch_run_fatal: Connection to XYZ port 22: Connection corrupted
> ```

Sadly also on 15.1 Beta (24B5055e)

**Comment by donghuachensu** [comment]
I updated my iMac to Sequoia 15.0.1 yesterday and got this: Bad packet length 205845566.
ssh_dispatch_run_fatal: Connection to x.x.x.x port 22: Connection corrupted
Very annoying. When will Apple fix this?

**Comment by myndzi** [comment]
Looking good for me too so far, hooray! It was pretty quick to ruin streams, so I'm hopeful.

**Comment by donghuachensu** [comment]
When will Sequoia 15.1 be released? Any estimate? Thanks!

**Comment by maramazza** [comment]
I confirm that with Sequoia 15.1 it is ok

**Comment by donghuachensu** [comment]
I just updated my iMac from Sequoia 15.01 to Sequoia Public Beta 15.1 (24B83) about 1.5 hours ago. So far so good for Terminal ssh connection (no disconnection error "ssh_dispatch_run_fatal").

**Comment by objective-see** [comment]
Close this, this has now been fixed (by Apple!) for a while!

Also, have added the following note on the website:

![Image](https://github.com/user-attachments/assets/49d77503-2b5f-468b-bd35-e4ddff92c18a)
